render_silhouette_regmarks.py

```diff
@@ -0,0 +1,132 @@
+	def drawRect(self, size, pos, name):
+		x, y = [pos * self.mm_to_user_unit for pos in pos  ]
+		w, h = [pos * self.mm_to_user_unit for pos in size ]
+		return inkex.Rectangle.new(x, y, w, h, id=name, style='fill: black;')
+		
+	#SVG line element generation routine
+	def drawLine(self, posStart, posEnd, name):
+		x1, y1, = [pos * self.mm_to_user_unit for pos in posStart]
+		x2, y2, = [pos * self.mm_to_user_unit for pos in posEnd  ]
+		line_style = 'stroke: black; stroke-width: '+str(self.REG_MARK_LINE_WIDTH_MM * self.mm_to_user_unit)+';'
+		return inkex.Line.new((x1, y1), (x2, y2), id=name, style=line_style)
+	
+	def effect(self):
+		reg_origin_X = self.options.regoriginx
+		reg_origin_Y = self.options.regoriginy
+		reg_width = self.options.regwidth if self.options.regwidth else int(self.svg.get("width").rstrip("mm")) - reg_origin_X*2
+		reg_length = self.options.reglength if self.options.reglength else int(self.svg.get("height").rstrip("mm")) - reg_origin_Y*2
+
+		if self.options.verbose == True:
+			inkex.base.InkscapeExtension.msg(gettext("[INFO]: page width ")+str(self.svg.get("width").rstrip("mm")))
+			inkex.base.InkscapeExtension.msg(gettext("[INFO]: page height ")+str(self.svg.get("height").rstrip("mm")))
+			inkex.base.InkscapeExtension.msg(gettext("[INFO]: regmark from document left ")+str(reg_origin_X))
+			inkex.base.InkscapeExtension.msg(gettext("[INFO]: regmark from document top ")+str(reg_origin_Y))
+			inkex.base.InkscapeExtension.msg(gettext("[INFO]: regmark to regmark spacing X ")+str(reg_width))
+			inkex.base.InkscapeExtension.msg(gettext("[INFO]: regmark to regmark spacing Y ")+str(reg_length))
+
+		# Create a new layer
+		layer = self.svg.add(inkex.Layer.new(self.LAYERNAME))
+	
+		# Create square in top left corner
+		layer.append(self.drawRect((self.REG_SQUARE_MM,self.REG_SQUARE_MM), (reg_origin_X,reg_origin_Y), 'TopLeft'))
+		
+		# Create group for top right corner
+		topRight = inkex.Group(id = 'TopRight', style = 'fill: black;')
+		# Create horizontal and vertical lines in group
+		top_right_reg_origin_x = reg_origin_X+reg_width
+		topRight.append(self.drawLine((top_right_reg_origin_x-self.REG_LINE_MM,reg_origin_Y), (top_right_reg_origin_x,reg_origin_Y), 'Horizontal'))
+		topRight.append(self.drawLine((top_right_reg_origin_x,reg_origin_Y), (top_right_reg_origin_x,reg_origin_Y + self.REG_LINE_MM), 'Vertical'))
+		layer.append(topRight)
+		
+		# Create group for top right corner
+		bottomLeft = inkex.Group(id = 'BottomLeft', style = 'fill: black;')
+		# Create horizontal and vertical lines in group
+		top_right_reg_origin_y = reg_origin_Y+reg_length
+		bottomLeft.append(self.drawLine((reg_origin_X,top_right_reg_origin_y), (reg_origin_X+self.REG_LINE_MM,top_right_reg_origin_y), 'Horizontal'))
+		bottomLeft.append(self.drawLine((reg_origin_X,top_right_reg_origin_y), (reg_origin_X,top_right_reg_origin_y - self.REG_LINE_MM), 'Vertical'))
+		layer.append(bottomLeft)
+		
+		# Keepout Marker #
```

Comment:
I didn't get the intention of those keepout markers. Are the needed for silhouette or intended as visual marks, but what exactly would they mark?